shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -582,19 +582,19 @@ public void onAttach(@NonNull Context context) {
     delegate.onAttach(context);
   }
 
+  @Override
+  public void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
I didn't quiet understand the bullet in the description on this change. Can you elaborate a bit more?

Reply:
I updated the bullet in the description slightly. In native Android, code assumes that state should be restored during or before `onCreate()`, so that that state can be accessed in `onCreate`. In a fragment or activity, this would all happen directly in the `onCreate` method, though that's not really applicable here. What is applicable is a `LifecycleObserver`, which would also expect to be able to rely on restored state in `onCreate` via Flutter's `OnSaveInstanceStateListener`. Since `onActivityCreated` gets called after `onCreate`, any plugin that uses `OnSaveInstanceStateListener` in combination with a `LifecycleObserver` would not have its restored state available in `onCreate`, as it ought to expect.

Reply:
Understood. Thanks for clarifying.